x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/analytics_list/use_columns.tsx

```diff
@@ -281,24 +281,25 @@ export const useColumns = (
   ];
 
   if (isManagementTable === true) {
-    if (spacesEnabled === true) {
-      // insert before last column
-      columns.splice(columns.length - 1, 0, {
-        name: i18n.translate('xpack.ml.jobsList.analyticsSpacesLabel', {
-          defaultMessage: 'Spaces',
-        }),
-        render: (item: DataFrameAnalyticsListRow) =>
-          Array.isArray(item.spaceIds) ? (
-            <JobSpacesList
-              spaceIds={item.spaceIds ?? []}
-              jobId={item.id}
-              jobType="data-frame-analytics"
-              refresh={refresh}
-            />
-          ) : null,
-        width: '90px',
-      });
-    }
+    // Note: this code path is commented because it is currently unreachable, it will need to be refactored to use the SpacesApi
```

Comment:
looks like this still needs changing. It's not possible to assign DFA jobs to spaces.

Reply:
Whoops, I don't know why I thought the DataFrameAnalyticsList was only used in one place, it's actually used in two places. That's what screwed me up.

Fixed in 265b85c and tested to make sure I could create + share a DFA job.

It's worth noting that I couldn't keep the SpacesContext at the JobsListPage level where it used to be, because that component re-renders a few times and somehow that winds up causing an infinite re-rendering loop when the SpacesContext is introduced there. So, instead, I added the SpacesContext wrapper within each tab section. Works like a charm, and the Spaces get reloaded (along with all the job data) when you switch tabs.

Reply:
there's an odd issue now on the DFA jobs list where the search bar loses focus after each character input. This is the cause of the failing functional test.

Reply:
Thanks for the pointer, that helped me figure out what was going on. The SpacesContext wrapper (along with all other components on this page) gets re-rendered multiple times with each key press. I discovered this by adding a console log inside of it 😄 As a side effect of that, the SpacesContext wrapper was creating a brand new context object each time it was rendered, which caused the wonky behavior of losing focus.

This was not a problem on the Saved Objects Management page, which is a bit simpler and only renders the SpacesContext wrapper once.

In d28bbda4 I changed the SpacesContext wrapper a bit to be more resilient, and because of that I was able to move it back to the top of the JobsListPage where it used to be. So I think it's all working perfectly now, you can take another look!

But FYI I think the JobsListPage has some problems that cause unnecessary re-rendering of components, might want to make a mental note to look into that in the future.